lib/mongoid/contextual/memory.rb

```diff
@@ -44,7 +44,7 @@ def delete
         deleted = count
         removed = map do |doc|
           prepare_remove(doc)
-          doc.as_document
+          doc.send(:as_attributes)
```

Comment:
The fewer methods exposed publicly on a document/model, the better. as_attributes is only ever used internally right now, so I don't want to get stuck not being allowed to change or get rid of it if I need to as part of a bug fix or refactor later on.